### PR TITLE
examples/evaluation: add PromptIter regression acceptance loop

### PR DESCRIPTION
**Pull request by shsaihdsaiudh**
## What changed

Adds a fully offline PromptIter regression loop that evaluates baseline and candidate instructions on separate training and held-out validation sets. It records failure attribution, per-case deltas, overfit and budget gate decisions, and JSON/Markdown audit reports.

The committed deterministic fixtures cover an ineffective candidate, a training-overfit candidate that is rejected, and a generalized candidate that passes all acceptance gates. No API key or external service is required.

## Why

PromptIter already generates candidate patches, but production adoption also needs reproducible held-out validation, explicit regression policy, and durable evidence explaining whether a candidate is safe to publish.

## Testing

- `cd examples/evaluation && go test ./promptiter/...`
- `cd examples/evaluation && go vet ./promptiter/regressionloop`
- `cd examples/evaluation && go run ./promptiter/regressionloop`
- `git diff --check`

## Notes for reviewers

The example introduces no public API. Candidate instructions use `evaluation/workflow/promptiter.PatchSet`; the deterministic candidate source models the boundary after PromptIter optimization so the acceptance and audit path remains reproducible without credentials.

Fixes #2003

### CODE REVIEW

File: examples/evaluation/promptiter/regressionloop/pipeline.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+func runPipeline(ctx context.Context, dataDir, outputDir string) (*optimizationReport, error) {
+	if err := ctx.Err(); err != nil {
+		return nil, err
+	}
+	loaded, err := loadInputs(dataDir)
+	if err != nil {
+		return nil, err
+	}
+	startedAt := time.Now().UTC()
```
Comment (Flash-LHR) [inline examples/evaluation/promptiter/regressionloop/pipeline.go]: Embedding `time.Now` in the tracked report makes each default run dirty `output/optimization_report.json`. Write default reports outside tracked output or use a stable fixture timestamp.
<details><summary>中文</summary>
在已提交报告中写入 `time.Now` 会让每次默认运行都改动 `output/optimization_report.json`。请将默认报告写到未跟踪目录，或为样例报告使用稳定时间。
</details>

Comment (shsaihdsaiudh) [reply examples/evaluation/promptiter/regressionloop/pipeline.go]: Fixed in c6b71c8b: wall-clock metadata was removed from committed reports, which are now byte-stable across repeated runs while deterministic per-evaluation latency remains audited.

File: examples/evaluation/promptiter/regressionloop/load.go [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+	}
+	if err := validateEvalSet("train", loaded.Train); err != nil {
+		return err
+	}
+	if err := validateEvalSet("validation", loaded.Validation); err != nil {
+		return err
+	}
+	if len(loaded.Config.Candidates) == 0 {
+		return errors.New("promptiter candidates are empty")
+	}
```
[the remark this answers is not in the document]
Comment (shsaihdsaiudh) [reply examples/evaluation/promptiter/regressionloop/load.go:100]: Fixed in c6b71c8b: candidate IDs and prompts are now validated with focused regression tests.

File: examples/evaluation/promptiter/regressionloop/README.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+Candidate instructions are represented as `evaluation/workflow/promptiter.PatchSet` values. The deterministic candidate source models the boundary after PromptIter's backward, aggregation, and optimizer stages while keeping the complete regression path runnable without credentials. Replacing it with a model-backed PromptIter engine changes candidate generation, not evaluation, gate, or reporting semantics.
+
+Every candidate is evaluated against both sets. Validation results are joined to the baseline by case ID and classified as new pass, new failure, improvement, regression, or unchanged. The acceptance gate requires a minimum held-out score gain, rejects every new failure and hard-case regression, and enforces call and estimated-token budgets. Therefore a prompt that improves all training cases but adds a memorized training-only rule is rejected, while a generalized routing and grounding improvement is accepted.
```
[the remark this answers is not in the document]
Comment (shsaihdsaiudh) [reply examples/evaluation/promptiter/regressionloop/README.md]: Fixed in c6b71c8b: the documented attribution categories now match the evaluator, and the per-candidate-round budget scope is explicit.

File: examples/evaluation/promptiter/regressionloop/report.go [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+			fmt.Fprintln(&output, "- Failure attribution:")
+			for _, category := range categories {
+				fmt.Fprintf(&output, "  - `%s`: %d\n", category, round.AttributionSummary[category])
+			}
+		}
+	}
+	fmt.Fprintln(&output)
+	fmt.Fprintln(&output, "## Accepted prompt")
+	fmt.Fprintln(&output)
+	fmt.Fprintf(&output, "```text\n%s\n```\n", report.AcceptedPrompt)
```
[the remark this answers is not in the document]
Comment (shsaihdsaiudh) [reply examples/evaluation/promptiter/regressionloop/report.go]: Fixed in c6b71c8b: per-case scores use three decimals and accepted prompts use a content-aware Markdown fence, covered by a regression test.

**Comment by shsaihdsaiudh** [comment]
Addressed all review feedback in c6b71c8b, including the three nitpicks: filepath.Join input paths, budget/hard-regression gate coverage, and byte-stable report metadata. Validation: go test ./... and go vet ./... in examples/evaluation; repeated report generation produced the same SHA-256.

File: examples/evaluation/promptiter/regressionloop/load.go [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+	for _, candidate := range loaded.Config.Candidates {
+		if strings.TrimSpace(candidate.ID) == "" {
+			return errors.New("candidate id is empty")
+		}
+		if strings.TrimSpace(candidate.Prompt) == "" {
+			return fmt.Errorf("candidate %q prompt is empty", candidate.ID)
+		}
+	}
```
[the remark this answers is not in the document]
Comment (shsaihdsaiudh) [reply examples/evaluation/promptiter/regressionloop/load.go:100]: Fixed in 43721b38: normalized candidate IDs are now tracked and duplicates are rejected; whitespace-normalized duplication has regression coverage.

File: examples/evaluation/promptiter/regressionloop/pipeline_test.go [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+	for name, cost := range map[string]costSummary{
+		"call budget":  {Calls: cfg.MaxCalls + 1},
+		"token budget": {EstimatedTokens: cfg.MaxEstimatedTokens + 1},
+	} {
+		t.Run(name, func(t *testing.T) {
+			decision := decideGate(cfg, baseline, evaluationSummary{Score: 1}, nil, cost)
+			require.False(t, decision.Accepted)
+			require.NotEmpty(t, decision.Reasons)
+		})
+	}
```
[the remark this answers is not in the document]
Comment (shsaihdsaiudh) [reply examples/evaluation/promptiter/regressionloop/pipeline_test.go]: Fixed in 43721b38: each budget subtest now asserts its exact call-count or estimated-token rejection reason.

**Comment by shsaihdsaiudh** [comment]
Addressed the two follow-up review comments in 43721b38. Validation: go test ./promptiter/regressionloop, go test ./..., and go vet ./... in examples/evaluation all pass.

**Comment by liuzengh** [comment]
感谢你对本 Issue 的贡献！

犀牛鸟活动的 Issue 提交阶段已经结束，本 PR 将关闭。再次感谢你的参与与付出！
